spec/dataframe_spec.rb

```diff
@@ -3266,6 +3266,15 @@
     end
   end
 
+  context "group_by" do
+    context "on a single row DataFrame" do
+      let(:df){ Daru::DataFrame.new({a: [1], b: [2]}) }
+      it "returns itself" do
```

Comment:
I am not sure it is the most reasonable behavior. Here is what happens if we have large DF:
```ruby
df = Daru::DataFrame.new(city: %w[Kyiv Kharkiv Odesa Kyiv Kharkiv Odesa], year: [2015, 2015, 2015, 2016, 2016, 2016], value: [1, 2, 3, 4, 5, 6])
# => #<Daru::DataFrame(6x3)>
#            city    year   value
#       0    Kyiv    2015       1
#       1 Kharkiv    2015       2
#       2   Odesa    2015       3
#       3    Kyiv    2016       4
#       4 Kharkiv    2016       5
#       5   Odesa    2016       6 
df.group_by(:city)
# => #<Daru::DataFrame(6x2)>
#                    year   value
# Kharkiv       1    2015       2
#               4    2016       5
#    Kyiv       0    2015       1
#               3    2016       4
#   Odesa       2    2015       3
#               5    2016       6 
```
Here is what (I believe) should happen with 1-row DF:
```ruby
df = Daru::DataFrame.new(city: %w[Kyiv], year: [2015], value: [1])
# => #<Daru::DataFrame(1x3)>
#        city  year value
#     0  Kyiv  2015     1 
df.group_by(:city)
# => #<Daru::DataFrame(1x2)>
#                    year   value
#    Kyiv       0    2015       1
```
(E.g. the same index/vectors change as with other groupings).

Reply:
Fair point, made it match the behaviour of other cases.